pkg/pod/creds_init_test.go

```diff
@@ -118,7 +118,123 @@ func TestCredsInit(t *testing.T) {
 		t.Run(c.desc, func(t *testing.T) {
 			names.TestingSeed()
 			kubeclient := fakek8s.NewSimpleClientset(c.objs...)
-			got, volumes, err := credsInit(images.CredsImage, serviceAccountName, namespace, kubeclient, volumeMounts, envVars)
+			got, volumes, err := credsInit(images.CredsImage, serviceAccountName, namespace, "", kubeclient, volumeMounts, envVars)
+			if err != nil {
+				t.Fatalf("credsInit: %v", err)
+			}
+			if got == nil && len(volumes) > 0 {
+				t.Errorf("Got nil creds-init container, with non-empty volumes: %v", volumes)
+			}
+			if d := cmp.Diff(c.want, got); d != "" {
+				t.Fatalf("Diff(-want, +got): %s", d)
+			}
+		})
+	}
+}
+
+func TestCredsInitGitHubApp(t *testing.T) {
+
+	sa := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{Name: serviceAccountName, Namespace: namespace},
+		Secrets: []corev1.ObjectReference{{
+			Name: "my-creds1",
+		}, {
+			Name: "my-creds2",
+		}},
+	}
+
+	for _, c := range []struct {
+		desc  string
+		want  *corev1.Container
+		objs  []runtime.Object
+		owner string
+	}{{
+		desc: "github organisation is foo so my-creds1 should be mounted into the container",
```

Comment:
worth adding a testcase for `owner: ""` to regression test the expected behaviour for that situation?